clients/ui/frontend/src/app/pages/modelCatalog/screens/ModelDetailsPage.tsx

```diff
@@ -58,15 +59,21 @@ const ModelDetailsPage: React.FC = () => {
   );
 
   const registerModelButton = () => {
-    if (
-      !modelRegistriesLoaded ||
-      modelRegistriesLoadError ||
-      !artifactLoaded ||
-      artifactsLoadError
-    ) {
+    if (!modelRegistriesLoaded || modelRegistriesLoadError) {
       return null;
     }
 
+    if (!artifactLoaded) {
+      return <Spinner size="xl" />; // Still loading artifacts
+    }
+
+    if (artifactsLoadError) {
+      return registerButtonPopover(
+        'Unable to load model artifacts',
+        'Model registration is unavailable due to an error loading model artifacts. Please try again later.',
+      );
+    }
```

Comment:
we should have this check before the `if(!artifactLoaded)` otherwise it will just show the spinner.